### DISCUSSION TITLE
Having a Request from the Public Page play as the next song up

### DISCUSSION THREAD

**Discussion by BillDewey** [AzuraCast/AzuraCast#7227] [Q&A]
Very much liking the interface, I have been using EverestCast and this makes things much simpler.  The one issue I am having is the AutoDJ Queue Length.  It appears I am forced to have 2 songs in the queue before a Request made from the Public Page will play.  The options in the Media Manager to play immediately or to queue as the next song are great, is there a work around to the queue length or have I just missed an option?  
Thank you very much for any and all responses and suggested, I have been searching the last couple of hours and if I have missed something please feel free to tell me so, just please point me in the right direction.

Thanks,

**Comment by photopixstore-ctrl** [comment]
Found the exact cause of this: `Queue::buildQueue()` has a hardcoded `max($station->backend_config->autodj_queue_length, 2)`, and the admin form field for "AutoDJ Queue Length" enforces `min=\"2\"` too - so there is currently no way to actually configure a queue depth of 1. Opened a minimal fix lowering that floor to 1 (AutoDJ always still keeps at least one committed slot): branch ready at https://github.com/AzuraCast/AzuraCast/compare/main...photopixstore-ctrl:AzuraCast:fix/autodj-queue-length-min-floor - couldnt open it as a PR from this account yet (seems to be a new-account PR limit), will convert it as soon as possible, or happy for a maintainer to cherry-pick directly. Also relevant: PR #8632 fixes a separate-but-related cause of request delay (the request_delay setting doing double duty for both anti-flood and playability timing).
